client/python/armada_client/event.py

```diff
@@ -0,0 +1,100 @@
+import enum
+import typing
+from dataclasses import dataclass
+
+import google.protobuf.timestamp_pb2 as timestamp_pb2
+
+from armada_client.armada import event_pb2, queue_pb2
+from armada_client.k8s.io.apimachinery.pkg.api.resource import (
+    generated_pb2 as resource_pb2,
+)
+
+
+class EventType(enum.Enum):
+    """
+    Enum for the event states.
+    """
+
+    submitted = "submitted"
+    queued = "queued"
+    duplicate_found = "duplicate_found"
+    leased = "leased"
+    lease_returned = "lease_returned"
+    pending = "pending"
+    running = "running"
+    unable_to_schedule = "unable_to_schedule"
+    failed = "failed"
+    succeeded = "succeeded"
+    reprioritized = "reprioritized"
+    cancelling = "cancelling"
+    cancelled = "cancelled"
+    terminated = "terminated"
+    utilisation = "utilisation"
+    ingress_info = "ingress_info"
+    reprioritizing = "reprioritizing"
+    updated = "updated"
+
+
+@dataclass
+class EventMessage:
+    """
+    EventMessage is the message type for the event stream.
+
+    Based on event_pb2.EventMessage
```

Comment:
IMO This is likely unacceptable; we cannot create manually curated files out of code that is generated from proto files. We cannot expect every commit to change an Armada API to require manual python code changes. 

Have you looked into if we can modify *how* the code generation happens for python, and perhaps create these dynamically?

Reply:
There are pretty much 0 options for generating the code differently, without major changes to how the proto stuff is layed out.

The problem is the fact that there's one proto message for each event type, and then G-Research have used the "one of" function, which is not very well supported in python, or golang. If you check the golang event.go file, it's virtually layed out the same as what I have done.

Reply:
https://github.com/Sharpz7/armada

Reply:
Actually, the golang event_util.go is a manual function they added to make working with events easier:

https://github.com/G-Research/armada/blob/master/pkg/api/event_util.go

Reply:
So casting this class onto the grpc class doesn't actually change anything about the object, so this will never break anything. It's purely here to make working with the class easier, and have everything written in a nicer fashion. 

If this is okay, I will add a docstring explaining this.

Reply:
We talked about this in slack. I have a better understanding of the struggles involved around this.

The idea moving forward is:
- This is fine, but not awesome. There is no awesome solution.
- We will document what code (outside of *.proto files) needs to be touched when an API is updated. -- this can be in a follow up / or just have an issue filed to do this later

Adam may, at his choice, spend some time in a timebox to generate this in CI -- but we think it'll be difficult to implement this in an consistently unbroken way.